nomad/datasource_nomad_job.go

```diff
@@ -0,0 +1,287 @@
+package nomad
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceJob() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceJobRead,
+		Schema: map[string]*schema.Schema{
+
+			"id": {
```

Comment:
Naming things `id`, unfortunately, really messes with our provider framework. `job_id` may be a better choice here.